lib/src/analytics.dart

```diff
@@ -69,15 +81,35 @@ abstract class Analytics {
   /// [shouldShowMessage] returns true
   String get toolsMessage;
 
+  /// Returns a map representation of the [UserProperty] for the [Analytics] instance
+  ///
+  /// This is what will get sent to Google Analytics with every request
+  Map get userPropertyMap;
+
+  /// Call this method when the tool using this package is closed
+  ///
+  /// Prevents the tool from hanging when if there are still requests
+  /// that need to be sent off
+  void close();
+
+  /// API to send events to Google Analytics to track usage
+  void sendEvent({
```

Comment:
I originally left this as `void` because I didn't want the tools using this package to have to resolve any non-2xx status codes if the request was unsuccessful; instead it would silently fail

I think we may need a discussion what we should do here

Reply:
You make a good point I hadn't considered that it may not be actionable :). On the other hand, why not return the response? The caller can always ignore it.